src/routes/events/containers/__tests__/EventsPage.spec.js

```diff
@@ -0,0 +1,188 @@
+    const mockDispatchLanguageChangeUrls = jest.fn()
+
+    const eventsPage = shallow(
+      <EventsPage events={events}
+                  location={location}
+                  languages={languages}
+                  language={language}
+                  dispatchLanguageChangeUrls={mockDispatchLanguageChangeUrls}/>
+    ).instance()
+
+    expect(mockDispatchLanguageChangeUrls.mock.calls).toHaveLength(1)
+    expect(mockDispatchLanguageChangeUrls).toBeCalledWith(eventsPage.mapLanguageToUrl, languages, {})
+  })
+
+  test('should dispatch on prop update with availableLanguages', () => {
+    const mockDispatchLanguageChangeUrls = jest.fn()
+
+    const wrapper = shallow(
+      <EventsPage events={[]}
+                  location={location}
+                  languages={languages}
+                  language={language}
+                  path={path}
+                  dispatchLanguageChangeUrls={mockDispatchLanguageChangeUrls}/>
+    )
+
+    expect(mockDispatchLanguageChangeUrls.mock.calls).toHaveLength(1)
+
+    wrapper.setProps({events: events, ...wrapper.props})
+
+    expect(mockDispatchLanguageChangeUrls.mock.calls).toHaveLength(2)
+    expect(mockDispatchLanguageChangeUrls).toBeCalledWith(
+      wrapper.instance().mapLanguageToUrl, languages, events[1].availableLanguages
+    )
+  })
+
+  test('should not dispatch on prop update', () => {
+    const mockDispatchLanguageChangeUrls = jest.fn()
+
+    const wrapper = shallow(
+      <EventsPage events={events}
+                  location={location}
+                  languages={languages}
+                  language={language}
+                  path={path}
+                  dispatchLanguageChangeUrls={mockDispatchLanguageChangeUrls}/>
+    )
+
+    let mockCalls = mockDispatchLanguageChangeUrls.mock.calls
+
+    wrapper.setProps({events: events, ...wrapper.props})
+
+    expect(mockDispatchLanguageChangeUrls.mock.calls).toHaveLength(mockCalls.length)
+
+    wrapper.setProps({...wrapper.props})
+
+    expect(mockDispatchLanguageChangeUrls.mock.calls).toHaveLength(mockCalls.length)
+  })
+
+  test('mapLanguageToUrl', () => {
+    const mockDispatchLanguageChangeUrls = jest.fn()
+
+    const mapLanguageToUrl = shallow(
+      <EventsPage events={events}
+                  location={location}
+                  languages={languages}
+                  language={language}
+                  path={path}
+                  dispatchLanguageChangeUrls={mockDispatchLanguageChangeUrls}/>
+    ).instance().mapLanguageToUrl
+
+    expect(mapLanguageToUrl('en')).toBe('/augsburg/en/events')
+    expect(mapLanguageToUrl('en', 1234)).toBe('/augsburg/en/events/1234')
+  })
+
+  const mockStore = configureMockStore([thunk])
+
+  test('should map state to props', () => {
+    const store = mockStore({
+      endpoint: new Payload(false),
+      endpoint1: new Payload(false),
+      router: {params: {location: 'augsburg', language: 'en', _: ''}}
+    })
+
+    const wrapper = mount(
```

Comment:
Expect is missing here. (I think this is still to do :))

Reply:
yeah right, you asked me to push this :P